### PR TITLE
Convert perf series function to react

### PR DESCRIPTION
**Pull request by camd**
This converts the ``PhSeries`` model to a generic JS model with ``fetch``.  We use this in the ``DetailsPanel`` in Treeherder, so need to convert it away from ``Angular``.

### CODE REVIEW

**Comment by camd** [comment]
@wlach : Ed is on PTO today, so no rush to review this today or anything.  And if you'd be up for some testing of the branch, that'd be awesome.  I tested graphs/compare/alerts myself, and things seem good, but don't want to miss anything.  :)

File: ui/helpers/url.js [hunk 1]
```diff
@@ -20,6 +20,22 @@ export const createQueryParams = function createQueryParams(params) {
   return `?${query.toString()}`;
 };
 
+// For API calls, we want arrays to be ``id=1&id=2&id=3`` not ``id=1,2,3``
+// Because the python ``getlist`` function on parameters can't handle the latter.
+export const createApiQueryParams = function createQueryParams(params) {
```
Comment (wlach) [inline ui/helpers/url.js]: Might want to consider using the `query-string` library here (could probably use for other things too): https://www.npmjs.com/package/query-string

File: ui/models/perfSeries.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    ).then((resp) => {
+      if (resp.ok) {
+        return resp.json();
+      }
+      return Promise.reject('No series data found');
+    });
+  }
+
+  static getReplicateData(params) {
+    params.value = 'perfherder-data.json';
```
Comment (wlach) [inline ui/models/perfSeries.js]: Let's not modify the value of `params` after it's passed in, instead just use the spread operator to modify it when passing it to createApiQueryParams.

File: ui/helpers/url.js (hunk 1 above)
Comment (camd) [reply ui/helpers/url.js]: Thanks for the tip!  I'll checkout out ``query-string``.  I'm curious how it deals with multiple values for the same "field".  :)

File: ui/models/perfSeries.js (hunk 2 above)
Comment (camd) [reply ui/models/perfSeries.js]: Ahh, good point.  Fixed.

**Comment by edmorley** [comment]
Yup, sounds good to me. The package is pretty small (https://bundlephobia.com/result?p=query-string@6.2.0) and is also actually used by taskcluster-client-web so we're pulling it into part of the bundle already.

**Pull request merged**